### PR TITLE
test(ci): require Responses API coverage in PR checks

### PR DESCRIPTION
**Pull request by serrrfirat**
## Summary

- run the complete feature-enabled `ironclaw_reborn_openai_compat` contract suite in crate CI
- move the two Responses route/temperature contracts from the root package into their owning crate
- require all 16 served Responses API E2E cases in Reborn PR CI
- share the same Responses E2E inventory with push-to-main coverage
- fail CI when the manifest drifts from the served Responses test modules
- reuse the prebuilt `openai-compat-beta` binary across the E2E modules
- restore the deterministic mock-LLM external-tool state machines and request-history endpoint removed by #5373 while their tests remained

## Why

The existing crate recipe enabled only `libsql`, so feature-gated route, workflow, and streaming contracts were omitted from normal crate CI. Responses E2E coverage was also split between the main coverage lane and tests that were not required on pull requests, allowing the inventories to drift.

Requiring the full served inventory exposed four dead external-tool tests immediately: their model scripts and request-history endpoint had been removed, so the mock returned a generic completed text response instead of a function call. This PR restores that deterministic fixture behavior without weakening or excluding the tests.

This makes the deterministic Responses contract a required PR gate without replacing Playwright UI coverage or relying on live canaries.

Related to #5990, which tracks the subsequent semantic-fidelity, lifecycle-safety, and external-tool phases. This PR does not close that issue.

## Validation

- `cargo test -p ironclaw_reborn_openai_compat --features openai-compat-beta,libsql` — 191 passed
- relocated contract tests — 14 passed
- exact served Responses manifest locally — 16 passed in 5.31s
- required Linux PR CI Responses step — 16 passed in 7s
- `python3 scripts/ci/check-reborn-responses-e2e-manifest.py` — 16/16 tests accounted for
- `bash scripts/ci/test-classify-test-scope.sh`
- Python syntax compilation for the checker, fixture, and changed E2E module
- workflow YAML syntax validation
- CI formatting check
- `git diff --check`

### CODE REVIEW

**Review by serrrfirat — COMMENTED** [review]
## Code Review (multi-agent)

**Intent:** Require complete deterministic Responses API contract and E2E coverage in pull-request CI while sharing manifests, detecting drift, and reusing the prebuilt Reborn binary.

**Mode:** normal — preflight found a 16-file, 33 KB diff with no generated/vendor concentration or stacked-diff signal.

**Coverage:** GitHub diff; 16 files (7 tests, 7 CI/scripts, 2 config/inventory); `diff_truncated=false`; no coverage limitations.

**Stats:** 3 findings from 3 raw findings across 3 files. Reviewers run: security, bugs, performance, tests, conventions, local-patterns, maintainability, approach. Reviewers failed: none. Body-only: 0.

### Tests

1. **Medium** Manifest drift rejection paths have no tests (`scripts/ci/check-reborn-responses-e2e-manifest.py:65-78`, confidence 100) — anchor: `scripts/ci/check-reborn-responses-e2e-manifest.py:65`
   The checker is only exercised against the valid repository manifest, so duplicate, missing, and unexpected selector rejection paths are not regression-tested.

2. **Low** Empty and duplicate feature manifests are untested (`scripts/live-canary/validate_reborn_binary_artifact.py:27-30`, confidence 100) — anchor: `scripts/live-canary/validate_reborn_binary_artifact.py:27`
   The parser rejects empty and duplicate features, but the new tests cover only a non-string array and a missing required feature.

### Maintainability

1. **Medium** Responses fixtures add three parallel hand-written state machines (`tests/e2e/mock_llm.py:1609-1745`, confidence 75) — anchor: `tests/e2e/mock_llm.py:1609`
   Three matchers repeat trigger detection, tool-result indexing, missing-tool handling, next-call construction, and final-summary construction, plus three dispatch branches.

File: scripts/ci/check-reborn-responses-e2e-manifest.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+def main() -> int:
+    manifest = manifest_node_ids()
+    manifest_set = set(manifest)
+    expected = expected_node_ids()
+
+    errors: list[str] = []
+    duplicates = sorted(
+        selector for selector, count in Counter(manifest).items() if count > 1
+    )
+    if duplicates:
```
Comment (serrrfirat) [inline scripts/ci/check-reborn-responses-e2e-manifest.py:65]: **Medium** — Manifest drift rejection paths have no tests.

The new checker is only executed against the valid repository manifest. No test mutates its inputs to prove that duplicate selectors, missing required selectors, and unexpected selectors actually fail, even though detecting those three drift modes is the checker stated purpose.

**Fix:** Add focused tests for duplicate, missing, and unexpected selectors using isolated fixture files.

File: scripts/live-canary/validate_reborn_binary_artifact.py [hunk 2]
```diff
@@ -14,6 +14,23 @@
...
+        features = [feature.strip() for feature in value.split(",")]
+    elif isinstance(value, list) and all(
+        isinstance(feature, str) for feature in value
+    ):
+        features = [feature.strip() for feature in value]
+    else:
+        raise ValueError(f"{field} must be a comma-separated string or string array")
+
+    if not features or any(not feature for feature in features):
+        raise ValueError(f"{field} contains an empty feature")
```
Comment (serrrfirat) [inline scripts/live-canary/validate_reborn_binary_artifact.py:28]: **Low** — Empty and duplicate feature manifests are untested.

The new feature parser explicitly rejects empty feature entries and duplicate features, but the added validator tests cover only a non-string array and a missing required feature.

**Fix:** Add tests covering an empty feature string and duplicate features for both supported manifest shapes.

File: tests/e2e/mock_llm.py [hunk 3]
```diff
@@ -1593,6 +1606,150 @@ def _find_named_tool_results(messages: list[dict], name: str) -> list[dict]:
     return [result for result in _find_tool_results(messages) if result.get("name") == name]
 
 
+def match_reborn_external_tool_response(
+    messages: list[dict],
+    has_tools: bool,
+) -> dict | None:
+    """Script a repeated caller-supplied Responses API tool loop."""
```
Comment (serrrfirat) [inline tests/e2e/mock_llm.py]: **Medium** — Responses fixtures add three parallel hand-written state machines.

The three new matchers repeat trigger detection, tool-result indexing, missing-tool handling, next-call construction, and final-summary construction, plus three dispatch branches. Additional scenarios will require more bespoke matchers in an already large fixture.

**Fix:** Collapse the matchers into a table-driven scripted-tool helper whose scenario data declares the trigger, calls, and final text, then use one dispatch branch.
